.castor/utils.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+use Castor\Attribute\AsContext;
+use Castor\Attribute\AsTask;
+use Castor\Context;
+use Symfony\Component\Process\Process;
+
+use function Castor\cache;
+use function Castor\capture;
+use function Castor\get_context;
+use function Castor\io;
+use function Castor\log;
+use function Castor\run;
+use function Castor\variable;
+
+#[AsTask(description: 'Displays some help and available urls for the current project')]
+function about()
+{
+    io()->section('About this project');
+
+    io()->comment('Run <comment>castor</comment> to display all available commands.');
+    io()->comment('Run <comment>castor about</comment> to display this project help.');
+    io()->comment('Run <comment>castor help [command]</comment> to display Castor help.');
+
+    io()->section('Available URLs for this project:');
+    $urls = [variable('root_domain'), ...variable('extra_domains')];
+
+    $payload = @file_get_contents(sprintf('http://%s:8080/api/http/routers', variable('root_domain')));
+    if ($payload) {
+        $routers = json_decode($payload, true);
+        $projectName = variable('project_name');
+        foreach ($routers as $router) {
+            if (!preg_match("{^{$projectName}-(.*)@docker$}", $router['name'])) {
+                continue;
+            }
+            if ("frontend-{$projectName}" === $router['service']) {
+                continue;
+            }
+            if (!preg_match('{^Host\\(`(?P<hosts>.*)`\\)$}', $router['rule'], $matches)) {
+                continue;
+            }
+            $hosts = explode('`, `', $matches['hosts']);
+            $urls = [...$urls, ...$hosts];
+        }
+    }
+    io()->listing(array_map(fn ($url) => "https://{$url}", $urls));
+}
+
+#[AsTask(description: 'Opens a shell (bash) into a builder container')]
+function builder(string $user = 'app')
+{
+    $c = get_context()
+        ->withTimeout(null)
+        ->withTty()
+        ->withEnvironment($_ENV + $_SERVER)
+        ->withQuiet()
+        ->withAllowFailure()
+    ;
+    docker_compose_run('bash', c: $c, user: $user);
+}
+
+#[AsContext(default: true)]
+function create_default_context(): Context
+{
+    $data = create_default_parameters() + [
```

Comment:
Should we name this function `create_default_variables()` since its retrieved later with the `variable()` function?